app/src/main/java/com/example/whereto/fragments/PlanTripFragment.java

```diff
@@ -1,21 +1,228 @@
 package com.example.whereto.fragments;
 
+import android.app.DatePickerDialog;
+import android.app.Dialog;
 import android.os.Bundle;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.DatePicker;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
+import com.daprlabs.cardstack.SwipeDeck;
 
+import com.example.whereto.Question;
+import com.example.whereto.QuestionsAdapter;
 import com.example.whereto.R;
+import com.google.android.material.slider.LabelFormatter;
+import com.google.android.material.slider.Slider;
+
+import java.text.NumberFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Currency;
 
 public class PlanTripFragment extends Fragment {
-    public PlanTripFragment() {
+    private SwipeDeck cardStack;
+    private ArrayList<Question> questionList;
+    EditText startDate;
+    EditText endDate;
+    DatePickerDialog datePickerDialog;
+
+    public float budget = 0;
+    public float radius = 0;
+    public String tripStart;
+    public String tripEnd;
+    public boolean food = false;
+    public boolean hotels = false;
+    public boolean tours = false;
+    public boolean athletic = false;
+    public boolean arts = false;
+    public boolean shopping = false;
+    public boolean nightlife = false;
+    public boolean beauty = false;
```

Comment:
It seems like all of these are the users preferences. Could we create a preferences object that holds all these values. Then we don't have so many manipulatable variables in this fragment.